src/main/java/gregtech/api/recipes/RecipeMap.java

```diff
@@ -261,6 +286,17 @@ private Recipe findByInputs(long voltage, List<ItemStack> inputs, List<FluidStac
         return null;
     }
 
+    @SuppressWarnings("unchecked")
+    @Nullable
+    private Recipe findByInputs(long voltage, List<ItemStack> inputs) {
+        for (Recipe recipe : recipeList) {
+            if (recipe.matches(false, inputs, Collections.EMPTY_LIST, Recipe.MatchingMode.ITEM_ONLY) && !recipe.getInputs().isEmpty()) {
```

Comment:
the `&& !recipe.getInputs().isEmpty()` is important without it machines starts processing from nothing. And I didn't really understand your sentence.

Reply:
Yeah it seems some words got lost when I was rewriting it. I meant that `matches` in items_only mode should not return true if provided item lists is empty as that usage does not make sense. And that would mean that there would be two similar function from which one is missing only `MatchingMode` and because they are private it would be correct to squash them in to one function - no need to consider API breaking (as it is not part of it).